cluster-autoscaler/core/scale_up.go

```diff
@@ -90,28 +117,43 @@ func ScaleUp(context *AutoscalingContext, unschedulablePods []*apiv1.Pod, nodes
 			continue
 		}
 
-		currentSize, err := nodeGroup.TargetSize()
+		currentTargetSize, err := nodeGroup.TargetSize()
 		if err != nil {
 			glog.Errorf("Failed to get node group size: %v", err)
 			continue
 		}
-		if currentSize >= nodeGroup.MaxSize() {
+		if currentTargetSize >= nodeGroup.MaxSize() {
 			// skip this node group.
 			glog.V(4).Infof("Skipping node group %s - max size reached", nodeGroup.Id())
 			continue
 		}
 
-		option := expander.Option{
-			NodeGroup: nodeGroup,
-			Pods:      make([]*apiv1.Pod, 0),
-		}
-
 		nodeInfo, found := nodeInfos[nodeGroup.Id()]
 		if !found {
 			glog.Errorf("No node info for: %s", nodeGroup.Id())
 			continue
 		}
 
+		nodeCPU, nodeMemory, err := getNodeCPUAndMemory(nodeInfo)
+		if err != nil {
+			glog.Errorf("Failed to get node resources: %v", err)
+		}
+		if context.MaxCoresTotal > 0 && nodeCPU > (context.MaxCoresTotal-coresTotal) {
```

Comment:
set a default value for MaxCoreTotal and MemTotal to some big values like
cpu = 5000 * 64 
mem = 5000 * 64 * 20

Reply:
Done